### ISSUE TITLE
Rule Change: no-case-declarations revisit #15851

### ISSUE THREAD

**Issue by DanKaplanSES** [eslint/eslint#17847]
# Revisit #15851

### What rule do you want to change?

no-case-declarations

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

Other

### Example code

```js
// Copied from https://github.com/eslint/eslint/issues/15851
// Examples of incorrect code for this rule

switch (foo) {
    case 1:
        let x = 1;
        break;    
}
```

### What does the rule currently do for this code?

Gives "Unexpected lexical declaration in case block."

### What will the rule do after it's changed?

No warning/error because `x` is not used while having an uninitialized state.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

The original issue is here: #15851 [It has one reply](https://github.com/eslint/eslint/issues/15851#issuecomment-1120318663) by @btmills:

> I could maybe see adding an "unless referenced only with in a single case" mode to this rule. The downside there is future changes could cause non-local errors. Generally, we try to build rules such that writing code can only introduce errors in that code. It can be unexpected to change code in one place and get an error in another. **"If I'm declaring in a case, wrap it in curlies" is a much simpler rule for developers to keep in mind.**

But this bold part may contradict another comment in [a related issue](https://github.com/eslint/eslint/issues/8567#issuecomment-300079966) by @not-an-aardvark:

> Thanks for the proposal.
> 
> I don't think this should be fixable because the autofixer would end up changing the behavior of the code, which is generally something we try to avoid. For example, the following code would start throwing an error if a block was added:
> 
> ```
> const x = 1;
> switch (x) {
>   case 1:
>     const foo = 1;
>   default:
>     console.log(foo);
> }
> ```

I'd like to reopen this for discussion. Personally, I think the current rule would be improved if #15851 were implemented as described. Is this a possibility?

**Comment by mdjermanovic** [comment]
Hi @DanKaplanSES, thanks for the issue!

I think the purpose of this rule is to avoid confusion about scoping in switch statements, both in the current code and when the code changes in the future. Therefore, I don't think the current behavior of this rule, which is to enforce  "If I'm declaring in a case, wrap it in curlies" to make the scopes clear, should be changed.

MDN also [recommends](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch#lexical_scoping) the same:

> whenever you need to use `let` or `const` declarations in a case clause, wrap it in a block.

**Comment by snitin315** [comment]
I agree with @mdjermanovic here, I don't think the behavior should be changed. Closing this issue as we won't be making any changes.
